### PR TITLE
Adding EVEX encoding support for CV path.

### PR DESCRIPTION
**Pull request by DeepakRajendrakumaran**
This set of changes adds `EVEX` encoding logic to remaining `emitOutput*` paths(not covered in previous 2 commits). With this commit, we should now be able to encode `avx512` instructions once those are added.

**Notes** : I have chosen to use the all encompassing `'emitOutputSimdPrefixIfNeeded'` instead  of`'emitOutputRexOrVexPrefixIfNeeded'` even in cases where we know `EVEX` will not be used for uniformity. Internally it will resolve to `'emitOutputRexOrVexPrefixIfNeeded'` etc. This extends to other cases like '`TakesSimdPrefix`' of '`TakesVexPrefix`' etc.

**TO DOs:**
1. We still require `kmask` support, register support for `zmm` register and upper(16-31) registers in order to use those features with AVX512. These will be added as separate commits.
2. We will have to modify `'TakesEvexPrefix'` logic. Currently has comments there explaining the nature of logic required there. This will have to be fleshed out once AVX512 instructions are added.
3. Some EVEX encoding logic like embedded broadcast etc which are not present for non AVX512. This needs to be added later
4. Move 'W' bit as opcode extension to an instruction table driven approach

@anthonycanino

### CODE REVIEW

File: src/coreclr/jit/emitxarch.cpp [hunk 1]
```diff
@@ -12752,13 +12752,13 @@ BYTE* emitter::emitOutputCV(BYTE* dst, instrDesc* id, code_t code, CnsVal* addc)
         dst += emitOutputByte(dst, 0x64);
     }
 
-    // Compute VEX prefix
-    // Some of its callers already add VEX prefix and then call this routine.
-    // Therefore add VEX prefix is not already present.
-    code = AddVexPrefixIfNeededAndNotPresent(ins, code, size);
+    // Compute VEX/VEX prefix
```
Comment (Wraith2) [inline src/coreclr/jit/emitxarch.cpp:12755]: `EVEX/VEX` ?

Comment (DeepakRajendrakumaran) [reply src/coreclr/jit/emitxarch.cpp:12755]: yes! Will fix it in the final  non draft version. Thanks for catching that

**Comment by BruceForstall** [comment]
@DeepakRajendrakumaran The avx512 test pipeline was error-free!

**Comment by DeepakRajendrakumaran** [comment]
> @DeepakRajendrakumaran The avx512 test pipeline was error-free!

Yaay!. The other pending PR will handle most of the fails but this is also needed to fix some other paths. I want to add some other stuff to this PR and should have it ready early this week.

File: src/coreclr/jit/emitxarch.cpp (hunk 1 above)
Comment (DeepakRajendrakumaran) [reply src/coreclr/jit/emitxarch.cpp]: Done!

**Comment by DeepakRajendrakumaran** [comment]
@BruceForstall @tannergooding This is ready for review now. This should be the last change required in this series(EVEX in output path)

**Comment by BruceForstall** [comment]
> @BruceForstall @tannergooding This is ready for review now. This should be the last change required in this series(EVEX in output path)

You should edit the top comment so it doesn't read "Draft PR to run tests" anymore, but is more descriptive of the changes.

**Comment by DeepakRajendrakumaran** [comment]
> > @BruceForstall @tannergooding This is ready for review now. This should be the last change required in this series(EVEX in output path)
> 
> You should edit the top comment so it doesn't read "Draft PR to run tests" anymore, but is more descriptive of the changes.

Have updated it with relevant info.

**Comment by BruceForstall** [comment]
[No asm diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=84858&view=ms.vss-build-web.run-extensions-tab) but a throughput regression on x64 of about 0.03%.

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**